### ISSUE TITLE
Missing logger accessor attributes

### ISSUE THREAD

**Issue by aks** [sds/mock_redis#198]
Code that uses `redis.logger` or `redis.logger = other_logger` fails when using `MockRedis`.  The emulation should include these attributes.

**Comment by sds** [comment]
Happy to accept a pull request!

**Comment by jdelStrother** [comment]
AFAICT redis-rb doesn't implement logger accessors -
```
irb(main):007:0> Redis.new.logger
Traceback (most recent call last):
        1: from (irb):7
Redis::CommandError (ERR unknown command `logger`, with args beginning with:)
```

\- I assume you're just asking for the ability to [log redis commands](https://github.com/redis/redis-rb/blob/6542934f01b9c390ee450bd372209a04bc3a239b/lib/redis/client.rb#L329-L351) as they're executed? eg:
```
irb(main):011:0> Redis.new(logger: Logger.new(STDOUT)).hgetall "foo"
D, [2021-05-12T16:29:05.939290 #76845] DEBUG -- : [Redis] command=HGETALL args="foo"
D, [2021-05-12T16:29:05.940348 #76845] DEBUG -- : [Redis] call_time=0.80 ms
=> {}
```


I started to implement it in mock-redis here: https://github.com/jdelStrother/mock_redis/commit/2ba1a14a648026ea058c7956e2d5b90a16023976 - though it's kind of tricky to do it cleanly. Redis has the advantage of pushing all commands through the `process` method, which gives a single entrypoint for logging. In my version I just attach logging to all public methods of the Database modules. In addition, some mock-redis methods are built on top of each other - eg [hmget](https://github.com/sds/mock_redis/blob/62ad38691f49db8e1a0e5398c6e5cbd1696cceca/lib/mock_redis/hash_methods.rb#L72-L75) uses hget internally, so a single call to hmget produces multiple logs like this:

```
D, [2021-05-12T16:35:57.356922 #77623] DEBUG -- : [MockRedis] command=HMGET args="key" "a" "b" "c"
D, [2021-05-12T16:35:57.356943 #77623] DEBUG -- : [MockRedis] command=HGET args="key" "a"
D, [2021-05-12T16:35:57.356943 #77623] DEBUG -- : [MockRedis] command=HGET args="key" "b"
D, [2021-05-12T16:35:57.356943 #77623] DEBUG -- : [MockRedis] command=HGET args="key" "c"
```

which would be fixable by refactoring hget into public & private parts - eg:

```ruby
def hget(key, field)
  _hget(key, field)
end
def hmget(key, *fields)
  fields.flatten.map { |f| _hget(key, f) }
end

private
def _hget(key, field)
  with_hash_at(key) { |h| h[field.to_s] }
end
```

since only the public methods are logged, but that's starting to sound like a lot of tedious work. Any suggestions on a better implementation?

**Comment by jdelStrother** [comment]
Actually scratch all that, https://github.com/jdelStrother/mock_redis/commit/dde6209ae70b8fa21332d0c808e2348cf6fd4c50 seems a much saner approach. I'll submit a PR for proper review.

**Comment by sds** [comment]
Implemented in #211.
